### DISCUSSION TITLE
Environment variables example not working

### DISCUSSION THREAD

**Discussion by iamsauravsharma** [community/community#25708] [Actions]
<p>So I was reading docs and came to part of environment variable <a href="https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables" rel="nofollow noopener">https://help.github.com/en/articles/virtual-environments-for-github-actions#environment-variables</a> . When I copy pasted given example and run actions why given example is not working?. It is displaying Hello World ! but it should have displayed Hello World Mona The Octocat !</p>

**Comment by judge2020** [comment]
<p>Environment variables are case-sensitive. Either the variables names in the <code>env</code> key need to be uppercase or the env substitution in the run key need to be lowercase.</p>

**Comment by coreywebber** [comment]
<p>Its not working for me either, I thought it was me, but not sure it is anymore. ex. of my yaml</p>
<p>name: build-test<br>
on: [pull_request]</p>
<p>jobs:<br>
  build_database:<br>
    runs-on: self-hosted<br>
    <br>
    steps:<br>
    - name: Hello world<br>
      run: echo Hello world $FIRST_NAME $middle_name $Last_Name!<br>
      env:<br>
        FIRST_NAME: Mona<br>
        middle_name: The<br>
        Last_Name: Octocat</p>
<p>Ex: of the output</p>
<p> Hello world</p>
<p>5s</p>
<p>Run echo Hello world $FIRST_NAME $middle_name $Last_Name!</p>
<p>echo Hello world $FIRST_NAME $middle_name $Last_Name!</p>
<p>shell: C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.EXE -command “. ‘{0}’”</p>
<p>env:</p>
<p>FIRST_NAME: Mona</p>
<p>middle_name: The</p>
<p>Last_Name: Octocat</p>
<p>Hello</p>
<p>world</p>
<p>!</p>

**Comment by edwardskrod** [comment]
<p>This is also not working for me.  I can’t use environment variables at all in my .yml.</p>

**Comment by edwardskrod** [comment]
<p>name: .NET Core</p>
<p># Trigger on push<br>
on: [push]</p>
<p>jobs:<br>
  build:<br>
  <br>
    runs-on: windows-latest</p>
<p>    steps:<br>
    - uses: actions/checkout@v1</p>
<p>    - name: Setup .NET Core<br>
      uses: actions/setup-dotnet@v1<br>
      with:<br>
        dotnet-version: ‘3.0.100’</p>
<p>    # This is not an official MsBuild action:  <a href="https://github.com/topics/msbuild-action" rel="nofollow noopener">https://github.com/topics/msbuild-action</a><br>
    - name: Setup MSBuild.exe<br>
      uses: warrenbuckley/Setup-MSBuild@v1</p>
<p>    # Print the variables to ensure things are working properly<br>
    - name: Print<br>
      env:<br>
        BUILDMODE: SideLoadOnly<br>
        CONFIGURATION: Release<br>
        DISTRIBUTIONURL: \ddfiles\team\xaml\user\edskrod<br>
        GENERATEAPPINSTALLERFILE: true<br>
        RUNTIMEIDENTIFIER: win-x86<br>
        TARGETPLATFORM: x86<br>
      run: echo $BUILDMODE $GENERATEAPPINSTALLERFILE $RUNTIMEIDENTIFIER $TARGETPLATFORM</p>
<p>Output:</p>
<p>Run echo $BUILDMODE $GENERATEAPPINSTALLERFILE $RUNTIMEIDENTIFIER $TARGETPLATFORM</p>
<p>2 echo $BUILDMODE $GENERATEAPPINSTALLERFILE $RUNTIMEIDENTIFIER $TARGETPLATFORM<br>
3 shell: C:\Program Files\PowerShell\6\pwsh.EXE -command “. ‘{0}’”<br>
4 env:<br>
5 DOTNET_ROOT: C:\hostedtoolcache\windows\dncs\3.0.100\x64<br>
6 BUILDMODE: SideLoadOnly<br>
7 CONFIGURATION: Release<br>
8 DISTRIBUTIONURL: \ddfiles\team\xaml\user\edskrod<br>
9 GENERATEAPPINSTALLERFILE: true<br>
10 RUNTIMEIDENTIFIER: win-x86<br>
11 TARGETPLATFORM: x86</p>

**Comment by jherico** [comment, marked as answer]
<p>Different shells have different syntax for accessing environment variables, and different OSes have different default shells.</p>
<p>Assuming an environment variable named FOO, the $FOO syntax works only on the bash shell.</p>
<p>If you’re running on windows, the default shell is powershell, and the syntax would be $env:FOO</p>
<p>Alternatively, if you want your workflow to support multiple OSes, and want to use environment variables in the commadn line without changes between OSes, then you should explicitly specify the shell as bash.</p>

**Comment by edwardskrod** [comment]
<p>Thanks Jherico!</p>
<p>I completely forgot about the OS.  The output is now correct:</p>
<p> Run echo $env:BUILDMODE $env:GENERATEAPPINSTALLERFILE $env:RUNTIMEIDENTIFIER $env:TARGETPLATFORM<br>
12  SideLoadOnly<br>
13  true<br>
14  win-x86<br>
15   x86</p>

**Comment by mikeBulava** [comment]
<p>This works for me:</p>
<pre><code class="lang-auto">- name: 'Update Variable'
  run: |
     echo "VAR1=${{env.VAR1}}-suffix" | Out-File -FilePath $Env.GITHUB_ENV -Encoding utf8 -Append
</code></pre>
<p>Credit to <a href="https://www.jamescroft.co.uk/setting-github-actions-environment-variables-in-powershell/" rel="noopener nofollow ugc">this article</a></p>
